doc/Programming-guide/Key-platform-features/Data-collection/index.md

```diff
@@ -5,4 +5,1175 @@ permalink: /:path/
+
+```json
+{  
+    "name":"EmptyLog",  
+    "namespace":"org.kaaproject.sample",  
+    "type":"record",  
+    "fields":[
+    ]
+}
+```
+
+* A simple log schema with the log level, tag, and message:
+
+```json
+{
+    "name":"LogData",
+    "namespace":"org.kaaproject.sample",
+    "type":"record",
+    "fields":[
+        {
+            "name":"level",
+            "type":{
+                "type":"enum",
+                "name":"Level",
+                "symbols":[
+                    "DEBUG",
+                    "ERROR",
+                    "FATAL",
+                    "INFO",
+                    "TRACE",
+                    "WARN"
+                ]
+            }
+        },
+        {
+            "name":"tag",
+            "type":"string"
+        },
+        {
+            "name":"message",
+            "type":"string"
+        }
+    ]
+}
+```
+
+## Adding log schema
+
+The default log schema installed for Kaa applications is empty. You can configure your own log schema using the Admin UI or
+[Admin REST API]({{root_url}}Programming-guide/Server-REST-APIs/ #TODO).
+For the purpose of this guide, we will use schema that is very close to the common log structure:
+* log level
+* tag
+* message
+
+```json
+{
+    "type":"record",
+    "name":"LogData",
+    "namespace":"org.kaaproject.kaa.schema.sample.logging",
+    "fields":[
+        {
+            "name":"level",
+            "type":{
+                "type":"enum",
+                "name":"Level",
+                "symbols":[
+                    "DEBUG",
+                    "ERROR",
+                    "FATAL",
+                    "INFO",
+                    "TRACE",
+                    "WARN"
+                ]
+            }
+        },
+        {
+            "name":"tag",
+            "type":"string"
+        },
+        {
+            "name":"message",
+            "type":"string"
+        }
+    ]
+}
+```
+
+In Admin UI as a tenant developer, you can create new log schemas for the application as follows:
+
+1. In the Log schemas window for the application, click Add schema.
+2. In the Add log schema window, create a log schema either by using the [schema form]({{root_url}}LINK_TO_AVRO_UI_FORMS_AVRO_UI_SCHEMA_FORM) or
+by uploading a schema in the [Avro](http://avro.apache.org/docs/current/spec.html) format from a file.<br/>
+
+![Add log schema](attach/add-log-schema.png) <br/>
+
+3. Click Add to save the schema.
+
+If you want to review the added Avro schema, open the Log schema details window by clicking the schema in the Log schemas window.
+
+![Log schema details](attach/log_schema_details.png)
+
+# Log appenders
+
+A log appender is a service utility which resides on the Operations service. This utility is responsible for writing logs (received by the Operations service
+from endpoints) to a single specific storage, as defined by the log appender's type. Kaa provides several default implementations of log appenders but
```

Comment:
... to a single specific storage, as defined by the log appender's type.

Is it possible to have more than one log appender (type) at the same time? If yes, this should be described additionally.